store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,314 @@
+}
+
+var DbDevicesCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID).
+			SetUnique(true),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldStatus, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldStatus),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldIDDataSha, Value: 1},
+		},
+		//nolint:staticcheck // SA1019
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldIDDataSha,
+			}, "_")).
+			SetUnique(true).
+			SetBackground(false),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldStatus, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		//nolint:staticcheck // SA1019
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldStatus,
+				dbFieldID,
+			}, "_")).
+			SetBackground(false),
+	},
+}
+
+var DbAuthSetsCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID).
+			SetUnique(true),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldDeviceID, Value: 1},
+			{Key: dbFieldIDDataSha, Value: 1},
+			{Key: dbFieldPubKey, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldDeviceID,
+				dbFieldIDDataSha,
+				dbFieldPubKey,
+			}, "_")).
+			SetUnique(true),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldIDDataSha, Value: 1},
+			{Key: dbFieldPubKey, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldIDDataSha,
+				dbFieldPubKey,
+			}, "_")).
+			SetUnique(true),
+	},
+}
+
+var DbLimitsCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldName, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldName).
+			SetUnique(true),
+	},
+}
+
+var DbTokensCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldExpTime, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldExpTime),
+	},
+}
```

Comment:
See my previous comments about how we manage tenant ID fields in the token collection.

Reply:
one of the reasons I want to keep it as a fielnd in Token struct. any way, if you insist we can talk.